Error running `npm install` on Windows when @stdlib/stdlib is a dependency

<!--lint disable first-heading-level-->

## Checklist

> Please ensure the following tasks are completed before filing a bug report.

-   [X] Read and understood the [Code of Conduct][code-of-conduct].
-   [X] Searched for existing issues and pull requests.

## Description

> Description of the issue.

Encountered an error on Windows when running `npm i` for the second time in a project which has @stdlib/stdlib as a dependency. 

## Related Issues

> Does this issue have any related issues?

No, I could not find any related issues.

The problem was caused, I believe, by line 282 in [this commit](https://github.com/stdlib-js/stdlib/commit/22ed99d5722620d49314b7d05a3c506ef25a0c26)

## Questions

> Any questions for reviewers?

No.

## Other

> Any other information relevant to this issue? This may include screenshots, references, stack traces, sample output, and/or implementation notes.

#### Reproduction

> What steps are required to reproduce the unexpected output?

In order to reproduce this bug, do the following:

**On Windows** (tested on Windows 10 Version 2004)
-   Create a new npm project with `@stdlib/stdlib` as a dependency in `package.json` 
```
mkdir bug-test && cd bug-test && npm init -y
edit package.json
# "dependencies": { "@stdlib/stdlib": "0.0.92"}
```
-   `npm install` - completes with no errors the first time 
-   `npm install` - second time installation throws the error

#### Expected Results

> What are the expected results?

The following results are expected:

- npm should install without an error

#### Actual Results

> What are the actual results?

The following are the actual results:

```text
npm ERR! code EINVALIDPACKAGENAME
npm ERR! Invalid package name "@stdlib/array\buffer": name can only contain URL-friendly characters
```

#### Environments

> What environments are affected (e.g., `Node v0.4.x`, `Chrome`, `IE 11`)? If Node.js, include the `npm` version, operating system, and any other potentially relevant platform information.

The following environments are affected:

-   npm on Windows (tested with npm v6.14.5, node v13.10.1, Windows 10 Version 2004)

<!-- <links> -->

[code-of-conduct]: https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md

<!-- </links> -->

--- a/tools/scripts/postinstall
+++ b/tools/scripts/postinstall
@@ -306,7 +306,7 @@ function main() {
 	debug( 'Modifying package meta data...' );
 	for ( i = 0; i < pkgs.length; i++ ) {
 		pdir = pkgs[ i ].slice( PKG_DIR.length+1 ); // +1 to account for trailing `/`
-		if ( pdir[ 0 ] !== '_' && pdir.split( '/' ).length === 1 ) {
+		if ( pdir[ 0 ] !== '_' && pdir.split( path.sep ).length === 1 ) {
 			meta.dependencies[ '@stdlib/'+pdir ] = 'file:./lib/node_modules/@stdlib/'+pdir;
 		}
 	}

